extendr-api/src/scalar/rbool.rs

```diff
@@ -0,0 +1,71 @@
+use crate::scalar::macros::*;
+use crate::*;
+use std::convert::TryFrom;
+
+/// Rbool is a wrapper for i32 in the context of an R's logical vector.
+///
+/// Rbool can have a value of 0, 1 or i32::MIN.
+///
+/// The value i32::MIN is used as "NA".
+///
+/// Rbool has the same footprint as an i32 value allowing us to use it in zero copy slices.
+pub struct Rbool(i32);
+
+impl Rbool {
+    gen_impl!(Rbool, i32);
+
+    pub fn t() -> Rbool {
+        Rbool(1)
+    }
+    pub fn f() -> Rbool {
+        Rbool(0)
+    }
```

Comment:
I really do not like `t` and `f`, but I guess there is no better option. `T` or `F` are against Rust naming convention, and something like `r#true` and `r#false` look ugly.

Reply:
I'm not a fan either.

If we can ditch Bool, we can re-purpose TRUE, FALSE and NA_BOOLEAN.

Reply:
I am tempted to do that now, I must admit.

Reply:
I'm in favor. Let's not add suboptimal naming to a new API because the optimal names clash with outdated parts. We're too early in the overall project development cycle.

As an alternative, though, you could also consider `true_value()` and `false_value()` or `get_true()` and `get_false()` or `logical_true()` and `logical_false()` or similar. Basically just combine true and false with one other word.